### PR TITLE
Fix weakmap dependency and add null property check

### PR DESCRIPTION
**Pull request by thoov**

### CODE REVIEW

File: addon/state-for.js [hunk 1]
```diff
@@ -27,11 +27,16 @@ export default function stateFor(stateName, propertyName) {
     return weakMaps[stateName];
   }
 
-  assert('The second argument must be a string', typeof(propertyName) === 'string');
+  assert('The second argument must be a string', typeof propertyName  === 'string');
 
   return computed(propertyName, function() {
     let propertyValue = this.get(propertyName);
 
+    assert(`The property '${propertyName}' must be an object, function, or array`,
```
Comment (thoov) [inline addon/state-for.js]: @stefanpenner This brings up an interesting question. Should we hard fail if the key that you pass in like so is null (or not a valid weakmap key):

```js
data: stateFor('state-name', 'my-key')
```

File: package.json [hunk 2]
```diff
@@ -38,14 +38,14 @@
...
-    "ember-weakmap": "0.3.0"
+    "ember-try": "0.0.6"
   },
   "keywords": [
     "ember-addon"
   ],
   "dependencies": {
-    "ember-cli-babel": "^5.1.3"
+    "ember-cli-babel": "^5.1.3",
+    "ember-weakmap": "0.4.0"
```
Comment (thoov) [inline package.json]: weakmap needs to be a dependency or an error will be thrown when the consumer uses the addon

File: addon/state-for.js (hunk 1 above)
Comment (stefanpenner) [inline addon/state-for.js]: so following existing ember patterns, `null` `undefined` keys, should just return `null` `undefined`

File: package.json (hunk 2 above)
Comment (stefanpenner) [inline package.json]: :+1:

File: addon/state-for.js (hunk 1 above)
Comment (thoov) [inline addon/state-for.js]: So if they pass in a number then the CP should just return that number?

Comment (stefanpenner) [inline addon/state-for.js]: > So if they pass in a number then the CP should just return that number?

Nope, it should assert/throw





> I think non-object keys other then undefined should likely TypeError (or maybe just assert)

Comment (thoov) [inline addon/state-for.js]: :+1:

**Comment by thoov** [comment]
@stefanpenner ok take a look now

**Comment by stefanpenner** [comment]
LGTM, merge when ready

**Pull request merged**